R/st_bonds.R

```diff
@@ -0,0 +1,194 @@
+#' )
+#' out <- suppressWarnings(
+#'   st_bonds(data, term = c(1, 2))
+#' )
+#'
+#' # Or set these environment variables (e.g. in .Renviron), then omit `data`
+#' data
+#'
+#' # Here we omit `data` and make the console output more verbose
+#' out <- st_bonds(term = c(1, 2), quiet = FALSE))
+#'
+#' # The data frame output helps you quickly explore and manipulate your results
+#' subset(out, st_name == "port" & arg_value == 2)
+#'
+#' # If you run out of memory, instead write output sequentially to disk
+#' path <- file.path(tempdir(), "st")
+#' dir.create(path)
+#' suppressWarnings(st_write_bonds(destdir = path, term = c(1, 2)))
+#' list.files(path)
+#'
+#' # Later you may read all results at once
+#' files <- list.files(path, full.names = TRUE)
+#' results <- read_csv(files, id = "arg_val", show_col_types = FALSE)
+#' results
+#'
+#' # The column `arg_val` helps you identify each run
+#' unique(
+#'   subset(results, grepl("term___2", arg_val), select = 1:3)
+#' )
+st_bonds <- function(data = st_data_paths(), ..., quiet = TRUE) {
+  st_df(data, asset_type = "bonds", ..., quiet = quiet)
+}
+
+#' @rdname st_bonds
+#' @export
+st_equity <- function(data = st_data_paths(), ..., quiet = TRUE) {
+  st_df(data, asset_type = "equity", ..., quiet = quiet)
+}
+
+#' @rdname st_bonds
+#' @export
+st_loans <- function(data = st_data_paths(), ..., quiet = TRUE) {
+  st_df(data, asset_type = "loans", ..., quiet = quiet)
+}
+
+#' @rdname st_bonds
+#' @export
+st_write_bonds <- function(data = st_data_paths(),
+                           destdir = tempdir(),
+                           ...,
+                           quiet = TRUE) {
+  st_write(data, asset_type = "bonds", destdir = destdir, ..., quiet = quiet)
+}
+
+#' @rdname st_bonds
+#' @export
+st_write_equity <- function(data = st_data_paths(),
+                            destdir = tempdir(),
+                            ...,
+                            quiet = TRUE) {
+  st_write(data, asset_type = "equity", destdir = destdir, ..., quiet = quiet)
+}
+
+#' @rdname st_bonds
+#' @export
+st_write_loans <- function(data = st_data_paths(),
+                           destdir = tempdir(),
+                           ...,
+                           quiet = TRUE) {
+  st_write(data, asset_type = "loans", destdir = destdir, ..., quiet = quiet)
+}
+
+st_df <- function(data, asset_type, ..., quiet = TRUE) {
+  args <- enlist_args(data, asset_type, ..., quiet = quiet)
+
+  args %>%
+    map(~ exec(st, !!!.x)) %>%
+    enframe(value = "st_result") %>%
+    restructure_st_df() %>%
+    unnest(st_result)
+}
+
+st_write <- function(data, asset_type, destdir = tempdir(), ..., quiet = TRUE) {
+  args <- enlist_args(data, asset_type, ..., quiet = quiet)
+
+  for (i in seq_along(args)) {
+    out <- exec(st, !!!args[[i]])
+    out <- unnest(out, st_result)
+    write_csv(out, file = path(destdir, glue("{names(args)[[i]]}.csv")))
+  }
+
+  invisible(data)
+}
+
+enlist_args <- function(data, asset_type, ..., quiet) {
```

Comment:
I think this is the most interesting part of this PR: Creating a list with as many elements as values are passed to one and only one of the arguments that are forwarded to `run_stress_test()` (via `...`). Each element has those arguments to be forwarded, plus the arguments to `st()` -- which is a thin wrapper of `run_stress_test()` with a signature that is easier to work with.